### PR TITLE
Various fixes for GeoServer GUI + JDBCConfig

### PR DESCRIPTION
**Pull request by n-lagomarsini**
This pull request contains various fixes for the GeoServer performances using JDBCConfig plugin.

### CODE REVIEW

**Comment by jodygarnett** [comment]
@n-lagomarsini can you address this before the freeze?

File: src/community/jdbcconfig/src/test/java/org/geoserver/jdbcconfig/internal/DbMappingsOfflineTest.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package org.geoserver.jdbcconfig.internal;
+
+public class DbMappingsOfflineTest {
+
+}
```
Comment (mbarto) [inline src/community/jdbcconfig/src/test/java/org/geoserver/jdbcconfig/internal/DbMappingsOfflineTest.java]: Is this class used at all?

File: src/community/jdbcconfig/src/test/java/org/geoserver/jdbcconfig/internal/QueryBuilderTest.java [hunk 2]
```diff
@@ -20,17 +20,19 @@
...
 import org.junit.Ignore;
+import org.junit.Test;
 import org.opengis.filter.Filter;
 
 /**
  * @author groldan
  * @author Kevin Smith, OpenGeo
  */
-@Ignore // Doesn't actually test anything so ignore.  Used for debugging.
+ // Doesn't actually test anything so ignore.  Used for debugging.
```
Comment (mbarto) [inline src/community/jdbcconfig/src/test/java/org/geoserver/jdbcconfig/internal/QueryBuilderTest.java]: Can we remove the comment? It seems to me that now some testing is executed.

File: src/main/src/main/java/org/geoserver/function/IsInstanceOfFactory.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import org.opengis.filter.expression.Literal;
+
+/**
+ * This class implements the {@link FunctionFactory} interface and can be used for creating a new {@link IsInstanceOf} function from the input
+ * parameters.
+ * 
+ * @author Nicola Lagomarsini geosolutions
+ */
+public class IsInstanceOfFactory implements FunctionFactory {
+
```
Comment (mbarto) [inline src/main/src/main/java/org/geoserver/function/IsInstanceOfFactory.java]: Isn't there any other FunctionFactory in GeoServer that we can add this FunctionFactory functions to?
If there is none, can we imagine to rename this to something more general and become a factory for more functions in the future?
I would avoid to have a factory for each single Function we need.
Something like CatalogFunctionFactory, or ConfigFunctionFactory, maybe?

File: src/main/src/main/java/org/geoserver/security/AbstractResourceAccessManager.java [hunk 4]
```diff
@@ -57,39 +50,10 @@ public LayerGroupAccessLimits getAccessLimits(Authentication user, LayerGroupInf
         return null;
     }
     
-    protected Catalog getCatalog() {
-        return (Catalog) GeoServerExtensions.bean("catalog");
-    }
-    protected SecureCatalogImpl getSecurityWrapper() {
-        return GeoServerExtensions.bean(SecureCatalogImpl.class);
-    }
-
```
Comment (mbarto) [inline src/main/src/main/java/org/geoserver/security/AbstractResourceAccessManager.java]: Are we sure we want to remove this methods for derived classes?

**Comment by n-lagomarsini** [comment]
Hi Jody,

yes, I would like to address this pull request before the feature freeze. I have already applied the changes requested by Kevin and Mauro.

**Comment by smithkm** [comment]
Thanks for the changes. It's OK by me and I'm really hoping to to see this make 2.7 myself.

**Comment by jodygarnett** [comment]
This merge broke the build, not sure if was handled as a merge or if you rebased first Kevin?

**Pull request merged**